test/testinfra.py

```diff
@@ -401,22 +407,55 @@ def update_status(revision, context, last, changes):
                     if status.get("description", NO_TESTING) == NO_TESTING:
                         baseline = 0
 
-                (priority, changes) = self.prioritize(status, labels, baseline)
+                (priority, changes) = self.prioritize(status, labels, baseline, context)
                 if update_status(revision, context, status, changes):
-                    results.append((priority, "pull-%d" % number, revision, "pull/%d/head" % number, context))
+                    results.append((priority, GithubPullTask("pull-%d" % number, revision,
+                                                             "pull/%d/head" % number, context)))
+
+        return results
+
+    def scan(self, update, context, except_context=False):
+        tasks = self.scan_for_pull_tasks(update, context, except_context)
```

Comment:
if we're refactoring this anyway, maybe it would make sense to use named variables instead of indexed tuples? .priority, .task